quake1_example/class/func_plat.gd

```diff
@@ -1,27 +1,46 @@
 extends QmapbspQuakeFunctionDoor
 class_name QmapbspQuakeFunctionPlat
+# FIXME: platforms need to automatically return to previous position after move end, not toggle or wait -1
+
+func _def_speed() -> String : return '150'
+func _def_lip() -> String : return '-8' # >O_O<
+func _no_linking() -> bool : return false
+func _get_angle() -> int : return -1
+func _can_create_trigger() -> bool : return true
+func _get_trigger_padding() -> Vector3 : return Vector3(-0.75, -0.5, -0.75)
+
 
 func _gen_aabb() :
 	var height : int = props.get('height', '0').to_int()
-	if height == 0 :
-		super()
-	else :
-		aabb.size.y = height
+	if height == 0 : super()
+	else : aabb.size.y = height
 	aabb.size.y *= -1
 	
-func _def_lip() -> String : return '-8' # >O_O<
-func _no_linking() -> bool : return false
+	
+func _set_primary_trigger_position(col : CollisionShape3D, aabb: AABB) :
+	super(col, aabb)
+	col.position.y -= aabb.size.y
+	
+	
 func _player_touch(p : QmapbspQuakePlayer, pos : Vector3, nor : Vector3) :
-	if open and nor.y > 0.9 :
+	if not open and _touch_horizontal(nor) :
```

Comment:
If the platform is "up"/open and player touches it's side, it will trigger it, so it returns back down